lib/factorial.rb

```diff
@@ -1,6 +1,27 @@
 # Computes factorial of the input number and returns it
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n), where n is the value of number. The loop will run number - 2 times, so the
+# amount of time the loop runs varies linearly based on the value of the input number.
+# Space complexity: O(1), because the method will store the same amount of data no matter
+# the value of the input number. Question - initially I wrote this method creating a new variable on line 19,
+# above lesser_number. The variable was called total, and the value was set to the value of the number. Instead of the loop
+# multiplying number by lesser_number and constantly changing the value of number to store the factorial value, total stored
+# this value. Would this be a better solution, so as not to change the value of number? Or is eliminating the use of another
+# variable more space-efficient?
```

Comment:
It's totally fair game (in any language, not just Ruby) to modify the input number. It's passed by value, not by reference, so any changes you make only affect the copy of the variable in this function.

You need two local variables - one to accumulate the result, and one to track the factors you're multiplying into the result. You can use the input number as one of those (either one), so you only need one additional local. You could call that 'total' or 'result' and accumulate the value into that local, or you could call it 'factor' or something and use the input number to accumulate the value. Either choice is fine, though accumulating into a new variable like 'total' is probably more readable.